dhis2/dhis2-core
# feat: job progress tracking for sending and sync jobs [DHIS2-12858]

### Summary
This PR adds `JobProgress` tracking to a number of "simpler" jobs that send or snyc something.
This means they essentially have only a single stage where some action is triggered.

### Manual Testing
* find UID of your system job in http://localhost:8080/api/jobConfigurations
* run the system job using `POST` http://localhost:8080/api/jobConfigurations/{uid}/execute
* check the running (if you can be fast enough) http://localhost:8080/api/scheduling/running/
* check the results http://localhost:8080/api/scheduling/completed/
* check detail results http://localhost:8080/api/scheduling/completed/{type}

Note that not all jobs updated in this PR were tested as some require SMS gateway or email server configuration.

### Example Outputs
http://localhost:8080/api/scheduling/completed/DATA_SET_NOTIFICATION
```json
[
  {
    "status":"SUCCESS",
    "completedTime":"2022-03-23T15:45:40.470",
    "startedTime":"2022-03-23T15:45:40.437",
    "description":"Dataset notification",
    "stages":[
      {
        "status":"SUCCESS",
        "completedTime":"2022-03-23T15:45:40.469",
        "startedTime":"2022-03-23T15:45:40.438",
        "description":"Sending scheduled dataset notifications",
        "totalItems":0,
        "items":[
          
        ],
        "duration":31,
        "complete":true
      }
    ],
    "jobId":"YvAwAmrqAtN",
    "duration":33,
    "complete":true
  }
]
```
http://localhost:8080/api/scheduling/completed/VALIDATION_RESULTS_NOTIFICATION
```json
[
  {
    "status":"SUCCESS",
    "completedTime":"2022-03-23T16:10:13.075",
    "startedTime":"2022-03-23T16:10:12.760",
    "description":"Validation result notification",
    "stages":[
      {
        "status":"SUCCESS",
        "completedTime":"2022-03-23T16:10:13.075",
        "startedTime":"2022-03-23T16:10:12.760",
        "description":"Sending validation result notifications",
        "totalItems":0,
        "items":[
          
        ],
        "duration":315,
        "complete":true
      }
    ],
    "jobId":"Js3vHn2AVuG",
    "duration":315,
    "complete":true
  }
]
```
